tasks/task_manager.hh

```diff
@@ -337,17 +352,22 @@ public:
         future<task_ptr> make_task(task::task_impl_ptr task_impl_ptr, task_info parent_d = task_info{}) {
             auto task = make_lw_shared<task_manager::task>(std::move(task_impl_ptr));
             foreign_task_ptr parent;
+            bool abort = false;
             if (parent_d) {
-                task->get_status().sequence_number = co_await _tm.container().invoke_on(parent_d.shard, [id = parent_d.id, task = make_foreign(task)] (task_manager& tm) mutable {
+                task->get_status().sequence_number = co_await _tm.container().invoke_on(parent_d.shard, [id = parent_d.id, task = make_foreign(task), &abort] (task_manager& tm) mutable {
                     const auto& all_tasks = tm.get_all_tasks();
                     if (auto it = all_tasks.find(id); it != all_tasks.end()) {
-                        it->second->add_child(std::move(task));
-                        return make_ready_future<uint64_t>(it->second->get_sequence_number());
+                        return it->second->add_child(std::move(task), abort).then([it] {
```

Comment:
Since you have access to the parent task here, via `it->second` there's no need to push `abort` to `add_child`.
Assuming we have a `task::abort_requested()` method that peers into _impl->_as,
we can call it here after adding the child.